test/e2e/basic_test.go

```diff
@@ -0,0 +1,140 @@
+// +build e2e
+
+package e2e
+
+import (
+	"testing"
+	"time"
+
+	"github.com/yahoo/athenz/clients/go/zms"
+	"github.com/yahoo/k8s-athenz-syncer/test/e2e/framework"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+func TestBasicRoleUpdate(t *testing.T) {
+	f := framework.Global
+	domain := zms.DomainName(f.RoleDomain)
+	roleName := zms.EntityName(f.RoleName)
+	roleResource := zms.ResourceName(f.RoleName)
+	role := zms.Role{
+		Name: roleResource,
+	}
```

Comment:
before adding the role to the domain, could we check if the CR exists? we could fail early before updating the ZMS.
```
		cr, exists, err := f.CRClient.GetCRByName(f.RoleDomain)
		if err != nil {
			t.Log("Error while finding athenzdomain resource from store")
			return false, nil
		}
		if !exists {
			t.Log("Did not find cr")
			return false, nil
		}
```